test/reduce/reduce_no_index.cpp

```diff
@@ -1,248 +1,148 @@
-
+#include <gtest/gtest.h>
 using namespace ck;
 
-static struct option long_options[] = {{"inLengths", required_argument, nullptr, 'D'},
-                                       {"reduceDimensions", required_argument, nullptr, 'R'},
-                                       {"scales", required_argument, nullptr, 'S'},
-                                       {"help", no_argument, nullptr, '?'},
-                                       {nullptr, 0, nullptr, 0}};
+struct ReduceParam
+{
+    bool do_verification{true};
+    bool propagateNan{false};
+    bool useIndex{false};
+    bool time_kernel{false};
+    bool do_dumpout{false};
+    int init_method{2};
+    float alpha{1.0f};
+    float beta{0.0f};
+    std::vector<size_t> inLengths{64, 4, 280, 82};
+    std::vector<int> reduceDims{0, 1, 2, 3};
+    ReduceTensorOp reduceOpId{ReduceTensorOp::AMAX};
+};
 
-class SimpleAppArgs
+std::vector<std::vector<int>> SetGenericReduceDim()
 {
-    private:
-    int option_index = 0;
+    return {{0, 1, 2, 3}, {0, 1, 2}, {0, 1, 3}, {0, 2, 3}, {1, 2, 3}, {0}, {1}, {2}, {3}};
+}
 
-    public:
-    std::vector<size_t> inLengths;
-    std::vector<int> reduceDims;
-    std::vector<float> scales;
+template <typename T>
+class ReduceWithIndexTest : public ::testing::Test
+{
+    protected:
+    using InDataType  = std::tuple_element_t<0, T>;
+    using AccDataType = std::tuple_element_t<1, T>;
+    using OutDataType = std::tuple_element_t<2, T>;
 
-    int data_type;
-    int init_method = 1;
+    static std::vector<ReduceParam> params;
 
-    public:
-    void show_usage(const char* cmd)
+    static void SetUpTestSuite()
     {
-        std::cout << "Usage of " << cmd << std::endl;
-        std::cout << "--inLengths or -D, comma separated list of input tensor dimension lengths "
-                     "(only 4-d tensor supported)"
-                  << std::endl;
-        std::cout << "--reduceDimensions or -R comma seperated list of dimension indexes to reduce "
-                     "(only 1 or 3 or 4 dimensions supported)"
-                  << std::endl;
-        std::cout << "--scales or -S, comma separated two float values for alpha and beta"
-                  << std::endl;
-        std::cout << "Arg1 -- data type (0: fp16, 1: fp32, 3: int8, 5: bp16, 6: fp64)" << std::endl;
-        std::cout << "Arg2 -- init method(0=no init, 1=single integer value, 2=scope integer "
-                     "value, 3=decimal value)"
-                  << std::endl;
+        // set testcase variables
+        ReduceParam set;
+        const auto setReduceDim = SetGenericReduceDim();
+
+        for(std::size_t i(0); i < setReduceDim.size(); ++i)
+        {
+            set.reduceDims = setReduceDim[i];
+            set.reduceOpId = ReduceTensorOp::AMAX;
+            params.emplace_back(set);
+            set.reduceOpId = ReduceTensorOp::MIN;
+            params.emplace_back(set);
+            set.reduceOpId = ReduceTensorOp::MAX;
+            params.emplace_back(set);
```

Comment:
I'd prefer if we have `reduceOpId` as a parameter to `Run` function. See comment below.